### PR TITLE
Enhance docs

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
Builds on #26. Please review that one first.

This is an attempt to enhance the docs

- Use Panel favicon
- Use Panel logo
- Use White/ Black header instead of Gray header
- Add admonitions
- Add copy code button
- Fix markdown lint issues

Addresses #22 . Fully addressing this would require a custom css file. That can be done later.
Addresses #24 by fixing the most obvious Markdown style issues.
addresses #28 by putting docs instructions into seperate DEVELOPER_GUIDE.md file.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-# Examples
+# Panel Chat Examples
```
Comment (MarcSkovMadsen) [inline docs/index.md:2]: Renaming to Panel Chat Examples will keep the "Panel Chat Examples" when scrolling. This is much better than "Examples" if you want to copy-paste screen shots and share on social media.

File: docs/index.md [hunk 2]
```diff
@@ -9,9 +9,10 @@ pip install hatch
 hatch run panel serve docs/examples/**/*.py --static-dirs thumbnails=docs/assets/thumbnails --autoreload
 ```
 
-Note the default installation is not optimized for GPU usage. To enable GPU support for local
-models (i.e. not OpenAI), install `ctransformers` with the proper backend and modify the
-scripts configs' accordingly, e.g. `n_gpu_layers=1` for a single GPU.
+!!! note
```
Comment (MarcSkovMadsen) [inline docs/index.md:12]: Add admonition. Looks better.

File: mkdocs.yml [hunk 3]
```diff
@@ -1,18 +1,19 @@
 site_name: Panel Chat Examples
-site_description: Example recipes for Panel ChatInterface.
+site_description: Example recipes for Panel Chat Components
```
Comment (MarcSkovMadsen) [inline mkdocs.yml:2]: I think this site if for Panel Chats in general. Not just Panels ChatInterface components.

**Review by ahuang11 — APPROVED** [review]
I appreciate all the improvements! Looks good to me!

**Review by ahuang11 — APPROVED** [review]

**Pull request merged**